core/include/detray/propagator/navigator.hpp

```diff
@@ -438,231 +515,174 @@ class navigator {
         }
 
-        const volume_type &volume =
-            _detector->volume_by_index(navigation.volume());
-        // Navigation state is broken/not initialized
-        if (navigation.trust_level() == navigation::trust_level::e_no_trust) {
-            initialize_kernel(propagation, volume);
-            return;
-        }
-
-        const auto &track = propagation._stepping();
-        // Update only the current candidate, or close neighbors if we miss the
-        // current candidate - do this only when the navigation state is still
-        // largely consistent (high trust)
+        // Update only the current candidate and the corresponding next target
+        // - do this only when the navigation state is still coherent
         if (navigation.trust_level() == navigation::trust_level::e_high or
-            navigation.candidates().size() == 1) {
-            while (not navigation.is_exhausted()) {
-
-                intersection_type &candidate = *navigation.next();
-                update_candidate(track, candidate);
-
-                // This is likely the next target
-                if (is_reachable(candidate, track)) {
-
-                    if (navigation.is_on_object(track, candidate)) {
-                        navigation.set_state(navigation::status::e_on_target,
-                                             candidate.index,
-                                             navigation::trust_level::e_full);
-                        // Release stepping constraint from actors
-                        propagation._stepping.release_step();
-                        // Set the next object we want to reach - might be end()
-                        ++navigation.next();
-                        if (not navigation.is_exhausted()) {
-                            navigation.run_inspector(
-                                "Update next (high trust):");
-                            // Ready the next candidate in line
-                            continue;
-                        } else {
-                            // There is no next candidate left - re-evaluate
-                            navigation.set_no_trust();
-                            navigation.run_inspector(
-                                "Update (high trust, kernel exhausted):");
-                            return;
-                        }
-                    }
-                    // we are not on the next object
-                    else {
-                        navigation.set_state(
-                            navigation::status::e_towards_object,
-                            dindex_invalid, navigation::trust_level::e_full);
-                        navigation.run_inspector("Update (high trust):");
-                    }
-                    // Don't sort again when coming from high trust
-                    return;
-                }
-                // Try next candidate
-                ++navigation.next();
+            navigation.n_candidates() == 1) {
```

Comment:
If you move towards the outer ring portal in the beampipe volume, you only have one candidate and it makes no sense to go into the other trust levels. Since its an 'or' and we stay in ```high trust``` most of the time, I hoped it would not impact performance that badly (praying at this point :D)